client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,360 @@
+          header: 'Set score',
+          fields: [
+            {
+              type: 'select',
+              label: Array.isArray(seats) ? 'Player' : 'Team',
+              options: players,
+              variable: 'player'
+            },
+            {
+              type: 'select',
+              label: this.get('roundLabel'),
+              options: rounds,
+              variable: 'round'
+            },
+            {
+              type: 'number',
+              label: 'Score',
+              variable: 'score'
+            }
+          ]
+        });
+        const seat = Array.isArray(seats) ? widgets.get(result.player) : seats[result.player][0];
+        let scores = seat.get(scoreProperty);
+        if(Array.isArray(scores))
+          scores = [...scores];
+        else
+          scores = [];
+        scores[result.round-1] = result.score;
+        await seat.set(scoreProperty, scores);
+      } catch(e) {}
+    }
+  }
+
+  get(property) {
+    if(property != '_totals')
+      return super.get(property)
+    else {
+      // First get total score for each relevant seat
+      const totals = [];
+      const seats = this.getIncludedSeats();
+      if(Array.isArray(seats)) {// Getting seat totals
+        for (const seat of seats) {
+          const score = seat.get(this.get('scoreProperty'));
+          const index = seat.get('index');
+          totals[index] = this.getTotal(score);
+        }
+        return totals
+      } else if (typeof seats == 'object') { // Getting team totals
+        const teamTotals = [null];
+        for (const team in seats) {
+          const seatsInTeam = widgetFilter(w => w.get('type') == 'seat' && seats[team].includes(w));
+          const seatsScores = seats[team].map(w => w.get(this.get('scoreProperty')));
+          const seatsTotals = asArray(seatsScores).map( s => this.getTotal(s) );
+          teamTotals.push(this.getTotal(seatsTotals));
+        }
+        return teamTotals;
+      }
+      return null; // Neither array nor object, return null.
+    }
+  }
+
+  readOnlyProperties() {
+    return new Set([...super.readOnlyProperties(), '_totals']);
+  }
+
+  // Return a modified 'seat' array or object including the seat widgets (not just the seat ids) to actually be used.
+  getIncludedSeats() {
+    let seats = this.get('seats');
+    if(typeof seats == 'string') // Allow "seats": "Seat1"
+      seats = asArray(seats);
+    if(Array.isArray(seats) || seats === null) // Scoreboard just using seats
+       return [...widgetFilter(w => w.get('type') == 'seat' && (this.get('showAllSeats') || w.get('player')) && (!seats || seats.includes(w.get('id'))))]
+    else if(typeof seats == 'object') { // Scoreboard using teams
+      const newSeats = {};
+      for (const team in seats) {
+        newSeats[team] = asArray(seats[team]);
+        newSeats[team] = [... widgetFilter(w => w.get('type') == 'seat' && (this.get('showAllSeats') || w.get('player')) && newSeats[team].includes(w.get('id')))];
```

Comment:
I would just do `asArray(seats[team]).includes(w.get('id'))` instead of having two rows but I'm notoriously cryptic so what do I know... :rofl:

Reply:
Done. I avoided this originally for just this reason, but now that I know that's OK, I'll go crazy. 🤣